tests/queries/0_stateless/02302_column_decl_null_before_defaul_value.sql

```diff
@@ -0,0 +1,51 @@
+select 'create table, column +type +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL) ENGINE=MergeTree() ORDER BY tuple();
```

Comment:
Also not really related, but in https://clickhouse.com/docs/en/sql-reference/statements/create/table the documentation says that the type of a column is optional:

```
CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
(
    name1 [type1] [NULL|NOT NULL] [DEFAULT|MATERIALIZED|EPHEMERAL|ALIAS expr1] [compression_codec] [TTL expr1],
    name2 [type2] [NULL|NOT NULL] [DEFAULT|MATERIALIZED|EPHEMERAL|ALIAS expr2] [compression_codec] [TTL expr2],
    ...
) ENGINE = engine
```

--> ```[type1]``` and ```[type2]```

Trying a CREATE TABLE statement w/o column types doesn't work. Would be cool if you could also fix that, while at it.

Reply:
Either type or default value expression should be there

Reply:
Column type is optional if default expression is specified. Type is deduced from the default expression in this case. AFAIK there's nothing to fix.